python/cudf/cudf/core/column/decimal.py

```diff
@@ -208,8 +212,32 @@ def __rtruediv__(self, other):
 
     def _binaryop(self, other: ColumnBinaryOperand, op: str):
         reflect, op = self._check_reflected_op(op)
-        other, other_cudf_dtype = self._normalize_binop_operand(other)  # type: ignore[assignment]
-        if other is NotImplemented:
+
+        # Inline _normalize_binop_operand functionality
+        if isinstance(other, ColumnBase):
+            if not isinstance(other, NumericalBaseColumn):
+                return NotImplemented
+            elif other.dtype.kind in "fb":
```

Comment:
I find this slightly easier to reason about. It relaxes the assumption that all types are single characters.
```suggestion
            elif other.dtype.kind in {"f", "b"}:
```